File: c# project/Dal/Models/Product.cs
using System;
using System.Collections.Generic;

namespace Dal.Models;

public partial class Product
{
    public int ProductId { get; set; }

    public string ProductName { get; set; } = null!;

    public int? CategoryId { get; set; }

    public int? CompanyId { get; set; }

    public string? DescriptionProduct { get; set; }

    public double Price { get; set; }

    public string? ImageUrl { get; set; }

    public int StockQuantity { get; set; }

    public DateTime? LastUpdated { get; set; }

    public string? Season { get; set; }

    public string? Size { get; set; }

    public string? Gender { get; set; }

    public virtual Category? Category { get; set; }

    public virtual Company? Company { get; set; }

    public virtual ICollection<OrdersDetail> OrdersDetails { get; set; } = new List<OrdersDetail>();
}

File: projectAngular/p/src/app/classes/product.ts
export class Product {
    public productId!: number;
    public productName!: string;
    public cName!: string;
    public companyId!: number;
    public descriptionProducts?: string;
    public price!: number;
    public imageUrl!: string;
    public stockQuantity!: number;
    public lastUpdated!: string;
    public company!: string;
    public season!: string;
    public size!: string;
    public gender!: string;
    public compName!: string;
    public ordersDetails?: [];

    constructor(product?: Product,) {
        if(product){
        this.productId = product.productId;
        this.productName = product.productName;
        this.cName = product.cName;
        this.companyId = product.companyId;
        this.descriptionProducts = product.descriptionProducts;
        this.price = product.price;
        this.imageUrl = product.imageUrl;
        this.stockQuantity = product.stockQuantity;
        this.lastUpdated = product.lastUpdated;
        this.company = product.company;
        this.season = product.season;
        this.size = product.size;
        this.gender = product.gender;
        this.compName = product.compName;
        this.ordersDetails = product.ordersDetails;
    }
}

}
